distribution/tools/entitlement-agent/src/main/java/org/elasticsearch/entitlement/instrumentation/MethodKey.java

```diff
@@ -9,28 +9,10 @@
 
 package org.elasticsearch.entitlement.instrumentation;
 
-import org.objectweb.asm.Type;
-
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
 import java.util.List;
 
 /**
  *
  * @param className the "internal name" of the class: includes the package info, but with periods replaced by slashes
  */
-public record MethodKey(String className, String methodName, List<Type> parameterTypes, boolean isStatic) {
-    /**
-     * @return a {@link MethodKey} suitable for looking up the given {@code targetMethod} in the entitlements trampoline
-     */
-    public static MethodKey forTargetMethod(Method targetMethod) {
-        Type actualType = Type.getMethodType(Type.getMethodDescriptor(targetMethod));
-        return new MethodKey(
-            Type.getInternalName(targetMethod.getDeclaringClass()),
-            targetMethod.getName(),
-            List.of(actualType.getArgumentTypes()),
-            Modifier.isStatic(targetMethod.getModifiers())
-        );
-    }
-
-}
+public record MethodKey(String className, String methodName, List<String> parameterTypes, boolean isStatic) {}
```

Comment:
Note that this is now a `List<String>` rather than a `List<Type>` since we don't want to force the user of this record to include ASM.

Reply:
FYI @jdconrad

Reply:
This makes sense to me. I just worry about mixing up the different ways to represent class names as strings, but tests should let us know if it's broken pretty quickly :)

Reply:
Yeah. We could either come up with our own data type, or else just live with this being stringly typed and rely on tests. We can't really pass `Class` objects around without because that would require the classes to be loaded and accessible, and we're not sure either of those will be the case.

Reply:
String is good IMO. I think we just complicate things more by trying to introduce something ASM will otherwise provide. We just need good tests.

Reply:
++
We _could_ introduce our own Type, and adapt to/from ASM Type, but I don't really see the need for this at this point. String is fine. Maybe add a comment/make a note that we could review this if we find it's too weak.